homework-submission/week4/my-app/src/App.js

```diff
@@ -0,0 +1,52 @@
+import React, { Component } from 'react';
+import './App.css';
+import Header from './components/Header';
+import MainForm from './components/MainInfo';
+import AddForm from './components/AddForm';
+import GetInfo from './services/GetMainInfo';
+
+class App extends Component {
+  state = {
+    mainInfo: [],
+  };
+
+  componentDidMount() {
+    GetInfo.getMainInfo().then(mainInfo =>
+      this.setState({
+        mainInfo,
+      }),
+    );
+  }
+
+  handleItemRemove = id => {
+    const newMainInfo = this.state.mainInfo.filter(item => item._id !== id);
```

Comment:
costly code
and
better to take a copy of the data -> [...this.state.mainInfo].filter(........